### PR TITLE
refactor: remove unused and outdated files

### PR DESCRIPTION
**Pull request by Junjiequan**
## Description
This PR aims to cleanup outdated CI/E2E configurations.

**Main fixes:**

- Moved all the oudated facility specific config files to the to_be_deleted folder.
- Removed the pipeline triggers for ESS and MAX dev environments on master branch pushes, as both pipelines have been inactive for a significant time.
- Removed the ESS-specific scichat-loopback microservice from the e2e test Docker Compose file
- Removed workflow that builds & push docker images based on manual release.
- Reduced the number of Cypress configuration files to a single cypress.config.ts.
- Fixed Jasmine type assertion falsy errors (e.g., Property 'toBeFalsy' does not exist on type 'Assertion').
- Fixed Cypress defineConfig type issues (e.g.,Object literal may only specify known properties, but 'lbBaseUrl' does not exist in type 'EndToEndConfigOptions'.).

## Motivation
The CI and testing setup has accumulated several unused and unmaintained configuration files, leading to increased difficulty in maintaining and debugging the system

## Fixes:
Please provide a list of the fixes implemented in this PR

* Items added

## Changes:
Please provide a list of the changes implemented by this PR

* changes made

## Tests included
- [ ] Included for each change/fix?
- [ ] Passing? (Merge will not be approved unless this is checked) 

## Documentation
- [ ] swagger documentation updated \[required\]
- [ ] official documentation updated \[nice-to-have\]

### official documentation info
If you have updated the official documentation, please provide PR # and URL of the pages where the updates are included

## Backend version
- [ ] Does it require a specific version of the backend
- which version of the backend is required:

## Summary by Sourcery

Refactor the project by removing unused and outdated files, updating docker-compose configurations, and optimizing CI workflows. Update service images and configurations to improve compatibility and performance. Remove unnecessary pipeline triggers and enhance test workflow settings.

Enhancements:
- Update Traefik image to version 2.10 in the docker-compose configuration.
- Switch MongoDB image to 'mongo:latest' due to ARM64 support issues with Bitnami MongoDB.
- Update SciChat Loopback and backend service image tags to 'latest' in the docker-compose configuration.
- Add health check for the backend service in the docker-compose configuration.
- Remove unused and outdated files from the CI and deployment configurations.

CI:
- Remove ESS and MAXIV pipeline triggers from the deploy workflow.
- Optimize memory and file handling settings for Elasticsearch in the test workflow.
- Update Cypress configuration file paths and base URLs in the test workflow.

### CODE REVIEW

File: CI/e2e/docker-compose.e2e.yaml [hunk 1]
```diff
@@ -19,60 +21,37 @@ services:
...
-    command:
-      [
-        "./wait-for-it.sh",
-        "backend:3000",
-        "--",
-        "nginx",
-        "-g",
-        "daemon off;"
-      ]
+      dockerfile: Dockerfile
```
Comment (nitrosx) [inline CI/e2e/docker-compose.e2e.yaml:50]: Are you sure that we do not need to wait for the service to come on line?

**Review by nitrosx — COMMENTED** [review]
Just one question.
The rest looks fine to me.

File: CI/e2e/docker-compose.e2e.yaml [hunk 2]
```diff
@@ -19,60 +21,37 @@ services:
...
-      ]
-    volumes:
-      - "./CI/ESS/e2e/.env:/home/node/app/.env"
-    depends_on:
-      mongodb:
-        condition: service_healthy
+
   backend:
     image: ghcr.io/scicatproject/backend-next:latest
     command: sh -c "node dist/main"
```
Comment (minottic) [inline CI/e2e/docker-compose.e2e.yaml:27]: I think the command is part of the image already and could be removed

File: CI/e2e/docker-compose.e2e.yaml [hunk 3]
```diff
@@ -1,14 +1,16 @@
 version: "3.4"
 services:
   reverse-proxy:
-    image: traefik:2.5
+    image: traefik:2.10
```
Comment (minottic) [inline CI/e2e/docker-compose.e2e.yaml:4]: I am not sure I understand what we need traefik for

File: CI/e2e/docker-compose.e2e.yaml (hunk 1 above)
Comment (minottic) [inline CI/e2e/docker-compose.e2e.yaml:50]: I think this line could be removed as `Dockerfile` should be the defautl

**Review by minottic — COMMENTED** [review]
as this PR seems to do some cleanup, I would suggest to have a look at [this](https://docs.github.com/en/actions/sharing-automations/reusing-workflows), at least for `deploy.yml`, maybe as food for thoughts for another PR (or at leat an issue). In this way we could reuse the deploy.yml workflow in all repos and import from a central place

File: CI/e2e/docker-compose.e2e.yaml (hunk 1 above)
Comment (Junjiequan) [reply CI/e2e/docker-compose.e2e.yaml:50]: I've removed the wait script from command line, as the proper approach is to use healthcheck and depends_on to do the job.

File: CI/e2e/docker-compose.e2e.yaml [hunk 4]
```diff
@@ -1,14 +1,16 @@
...
-    image: traefik:2.5
+    image: traefik:2.10
     command: --api.insecure=true --providers.docker=true --entrypoints.web.address=:80
     ports:
       - 8080:80
     volumes:
       - /var/run/docker.sock:/var/run/docker.sock
   mongodb:
-    image: "bitnami/mongodb:latest"
+    ## As of 2024-10-22 bitnami mongoDB still do not support ARM64
```
Comment (minottic) [inline CI/e2e/docker-compose.e2e.yaml:11]: I think mongo is the official one anyway

**Review by minottic — APPROVED** [review]
Thanks!

File: CI/e2e/docker-compose.e2e.yaml (hunk 3 above)
Comment (nitrosx) [reply CI/e2e/docker-compose.e2e.yaml:4]: Let's leave it as it is for now and postpone the investigation.
I created an issue regarding this: #1634

**Review by nitrosx — APPROVED** [review]

**Pull request merged**